netlify-plugin-cloudinary/src/index.ts

```diff
@@ -257,49 +257,51 @@ export async function onBuild({
     await Promise.all(
       CLOUDINARY_ASSET_DIRECTORIES.map(
         async ({ inputKey, path: defaultPath }) => {
-          let mediaPaths = inputs[inputKey as keyof Inputs] || defaultPath;
-
-          // Unsure how to type the above so that Inputs['privateCdn'] doesnt mess up types here
+          try {
+            let mediaPaths = inputs[inputKey as keyof Inputs] || defaultPath;
+
+            // Unsure how to type the above so that Inputs['privateCdn'] doesnt mess up types here
+
+            if (!Array.isArray(mediaPaths) && typeof mediaPaths !== 'string') return;
+
+            if (!Array.isArray(mediaPaths)) {
+              mediaPaths = [mediaPaths];
+            }
+
+            mediaPaths.forEach(async mediaPath => {
+              const cldAssetPath = `/${path.join(PUBLIC_ASSET_PATH, mediaPath)}`;
+              const cldAssetUrl = `${host}${cldAssetPath}`;
+
+              const { cloudinaryUrl: assetRedirectUrl } = await getCloudinaryUrl({
```

Comment:
one thing im concerned about here is similar to a previous comment

if `getCloudinaryUrl` throws an error here - wouldn't it throw an error for the entire `try/catch`, thus preventing the plugin from continuing to upload other images in the array?

wouldn't it make more sense for the try to wrap the `getCloudinaryUrl` call alone?